src/services/permissions-checker.js

```diff
@@ -43,105 +40,20 @@ class PermissionsChecker {
-      return filtredConditions[0];
-    }
-
-    // During the tree travel, check if `conditions` & `aggregator` match with expectations
-    return filtredConditions.length === expectedConditionFilters.conditions.length
-      && (aggregator === expectedConditionFilters.aggregator || aggregator === 'and')
-      ? { aggregator, conditions: filtredConditions }
-      : null;
-  }
-
-  // Check if `expectedConditionFilters` at least contains a definition of `actualConditionFilters`
-  static _isConditionFromScope(actualFilterCondition, expectedFilterConditions) {
-    return expectedFilterConditions.filter((expectedCondition) =>
-      expectedCondition.value === actualFilterCondition.value
-      && expectedCondition.operator === actualFilterCondition.operator
-      && expectedCondition.field === actualFilterCondition.field).length > 0;
-  }
-
-  async _isScopeValid(permissionInfos, scope) {
-    const expectedConditionFilters = PermissionsChecker
-      ._computeConditionFiltersFromScope(permissionInfos.userId, scope);
-
-    // Find aggregated condition. filtredConditions represent an array of conditions that were
-    // tagged based on if it is present in the scope
-    const isScopeAggregation = (aggregator, conditions) => PermissionsChecker
-      ._isAggregationFromScope(aggregator, conditions, expectedConditionFilters);
-
-    // Find in a condition correspond to a scope condition or not
-    const isScopeCondition = (condition) => PermissionsChecker
-      ._isConditionFromScope(condition, expectedConditionFilters.conditions);
-
-    // Perform a travel to find the scope in filters
-    const scopeFound = await this.baseFilterParser.perform(
-      permissionInfos.filters,
-      isScopeAggregation,
-      isScopeCondition,
-    );
-
-    // In the case of only one expected condition, server will still send an aggregator which will
-    // not match the request. If one condition is found and is from scope then the request is valid
-    const isValidSingleConditionScope = !!scopeFound
-    && expectedConditionFilters.conditions.length === 1;
-
-    const isSameScope = !!scopeFound
-    && scopeFound.aggregator === expectedConditionFilters.aggregator
-    && !!scopeFound.conditions
-    && scopeFound.conditions.length === expectedConditionFilters.conditions.length;
-
-    return isValidSingleConditionScope || isSameScope;
-  }
-
-  async _isCollectionBrowseAllowed(collectionPermissions, permissionInfos, scope) {
-    if (!collectionPermissions
-      || !permissionInfos
-      || !PermissionsChecker
-        ._isPermissionAllowed(collectionPermissions.browseEnabled, permissionInfos.userId)) {
-      return false;
-    }
-
-    if (!scope) return true;
-
-    try {
-      return this._isScopeValid(permissionInfos, scope);
-    } catch (error) {
-      this.logger.error(error);
-      return false;
-    }
+  // eslint-disable-next-line class-methods-use-this
+  async _isCollectionBrowseAllowed(collectionPermissions, permissionInfos) {
```

Comment:
[MISC] Can't you make it a static method and call it as the others: `PermissionsChecker._isCollectionBrowseAllowed`?

(that would dismiss the eslint rule removal)